lib/payload/scanner/SSL_TLS/engine.py

```diff
@@ -0,0 +1,674 @@
+                    supportedProtocols.append(p)
+                    #Remote service supports sslv2
+            else:
+                responseProtocol = tls.responseProtocol(response)
+                contentType = tls.contentType(response)
+                if (contentType[0] == 'handshake' and contentType[1] == 'server_hello') and (responseProtocol == p):
+                    supportedProtocols.append(p)
+                    #Remote service supports: TLS1, TLS1.1,TLS1.2, SSLV3
+        tls.closeConnection()
+    return supportedProtocols
+
+def enumCiphers(protocol):
+    supportedCiphers = Set()
+    if protocol == 'SSLv2':
+        cipherList = tls.ssl2Ciphers()
+        sock = tls.connect()
+        response = tls.doClientHello(protocol, cipherList)
+        if response:
+            cert_len = struct.unpack('!H', response[7:9])[0]
+            cipher_spec_len = struct.unpack('!H', response[9:11])[0]
+            start = 13 + cert_len
+            cipher_spec = response[start:start+cipher_spec_len]
+            for cipher in chunks(cipher_spec, 3):
+                supportedCiphers.add(cipher)
+        else:
+            pass
+            #No server hello received for SSLv2! (check client hello)
+        tls.closeConnection()
+    else:
+        cipherList = tls.tlsCiphers()
+    serverHelloCipher = 1
+    while serverHelloCipher > 0:
+        for c in tls.tlsCiphers():
+            sock = tls.connect()
+            response = tls.doClientHello(protocol, cipherList)
+            if response:
+                contentType = tls.contentType(response)
+                if contentType[0] == 'handshake' and contentType[1] == 'server_hello':
+                    helloCipher = tls.serverHelloCipher(response)
+                    if helloCipher != "":
+                        if helloCipher in supportedCiphers:
+                            serverHelloCipher = 0
+                            break
+                        else:
+                            supportedCiphers.add(helloCipher)
+                            cipherList = removeFromCipherList(cipherList, helloCipher)
+                elif contentType[0] == 'alert' and contentType[1] == 'handshake_failure':
+                    serverHelloCipher = 0
+                    break
+                else:
+                    serverHelloCipher = 0
+                    break
+            else:
+                serverHelloCipher = 0
+                break
+            tls.closeConnection()
+    return supportedCiphers
+
+def processTarget(target, internal=False):
+    global tls
+    SSLdata = dict()
+    if not re.match(r'^.*[:]+[0-9]{1,5}$', target):
+        target = target + ":443"
+        #No port specified, setting a default one 443"
+    host = target.split(':')
+    if not internal:
+        pass
+        #Processing target
+    if reachable(host[0], host[1]):
+        tls = TLS(host[0], host[1])
+        protocols = enumProtocols()
+    #    print protocols
```

Comment:
I think that these print statements comments can be removed. Let me know if you think otherwise!!

Reply:
I kept them for debugging while developing the module, forgot to remove